src/opm/parser/eclipse/EclipseState/Schedule/Well/Well.cpp

```diff
@@ -702,12 +702,11 @@ bool Well::updateConnections(std::shared_ptr<WellConnections> connections_arg, b
 }
 
 bool Well::updateConnections(std::shared_ptr<WellConnections> connections_arg, const ScheduleGrid& grid) {
-    bool update = this->updateConnections(connections_arg, false);
-    if (this->pvt_table == 0 && !this->connections->empty()) {
+    const bool update = this->updateConnections(connections_arg, false);
+    if (update && this->pvt_table == 0 && !this->connections->empty()) {
```

Comment:
I guess you are fixing a bug with this; but to me it looks suspicious. I would like to be able to request an update (with `return true`) for a pure pvtnum update. Also - return an unecessary `true` should only waste some memory.

So - all in all I guess it would be good to discuss the problem you are fixing.

Reply:
I think I see your point, that the pvtnum cell property has changed and then must call updateConnections() to make any defaulted well pvtnum update correctly?

Sounds dangerous any which way to me. Either we must not update the pvtnum prop after creating the well/connection objects (this would work correctly with my proposal), or we must updateConnections() every time we change pvtnum. The second certainly is not obvious and sounds hard to enforce! If we must change pvtnum (or other props that have treatment like this), my proposal cannot be used. Perhaps we should rather not store pvtnum in the well, instead keeping it at zero, making the accessor fetch it when we need it? I.e. move the logic here to pvt_table_number()?